lightning/src/ln/channelmanager.rs

```diff
@@ -1320,6 +1330,7 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 				msg: update
 			});
 		}
+		self.pending_events.lock().unwrap().push(events::Event::ChannelClosed { channel_id: *channel_id,  err: ClosureDescriptor::ForceClosed });
```

Comment:
We should split out this error code from the `handle_error` -> `force_close_channel_with_peer` call path, and include the error message from the counterparty in that case.

Reply:
What the split off would serve ? Not generating a `ForceClosed` in case of `handle_error` or introduce another `ClosureDescriptor` to cover specifically `handle_error` ? Added the error message from counterparty.